precompiles/outposts/osmosis/setup_test.go

```diff
@@ -0,0 +1,75 @@
+// Copyright Tharsis Labs Ltd.(Evmos)
+// SPDX-License-Identifier:ENCL-1.0(https://github.com/evmos/evmos/blob/main/LICENSE)
+
+package osmosis_test
+
+import (
+	"testing"
+
+	"github.com/evmos/evmos/v15/precompiles/outposts/osmosis"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/evmos/evmos/v15/testutil/integration/evmos/factory"
+	"github.com/evmos/evmos/v15/testutil/integration/evmos/grpc"
+	testkeyring "github.com/evmos/evmos/v15/testutil/integration/evmos/keyring"
+	"github.com/evmos/evmos/v15/testutil/integration/evmos/network"
+	"github.com/evmos/evmos/v15/x/evm/statedb"
+
+	"github.com/stretchr/testify/suite"
+)
+
+const (
+	portID    = "transfer"
+	channelID = "channel-0"
+)
+
+type PrecompileTestSuite struct {
+	suite.Suite
+
+	network     *network.UnitTestNetwork
+	factory     factory.TxFactory
+	grpcHandler grpc.Handler
+	keyring     testkeyring.Keyring
+
+	stateDB *statedb.StateDB
+
+	precompile *osmosis.Precompile
+}
+
+func (s *PrecompileTestSuite) SetupTest() {
+	keyring := testkeyring.New(1)
+	integrationNetwork := network.NewUnitTestNetwork(
+		network.WithPreFundedAccounts(keyring.GetAllAccAddrs()...),
+	)
+	grpcHandler := grpc.NewIntegrationHandler(integrationNetwork)
+	txFactory := factory.New(integrationNetwork, grpcHandler)
+
+	headerHash := integrationNetwork.GetContext().HeaderHash()
+	stateDB := statedb.New(
+		integrationNetwork.GetContext(),
+		integrationNetwork.App.EvmKeeper,
+		statedb.NewEmptyTxConfig(common.BytesToHash(headerHash.Bytes())),
+	)
```

Comment:
would make sense to also add a util for this on the integration test utils? Or why is this manually defined here? I would've expected this to be part of the network setup

Reply:
This is for this specific tests that needed to receive teh statedb on their params. The network setup does this internally, it just does not expose this. But maybe the UnitTesting implementation of the network interface could have a method for returning the current statedb object

Reply:
Just added that and updated the tests